### PR TITLE
:seedling: Fix nil point error in metal3remediation_manager

### PR DESCRIPTION
**Pull request by zhouhao3**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: baremetal/metal3remediation_manager_test.go [hunk 1]
```diff
@@ -574,6 +574,12 @@ var _ = Describe("Metal3Remediation manager", func() {
 			},
 			RemediationTypeSet: false,
 		}),
+		Entry("Remediation strategy is nil should return empty string", testCaseGetRemediationType{
+			Metal3Remediation: &infrav1.Metal3Remediation{
+				Spec: infrav1.Metal3RemediationSpec{},
+			},
+			RemediationTypeSet: false,
+		}),
```
Comment (furkatgofurov7) [inline baremetal/metal3remediation_manager_test.go:582]: This looks like a duplicate of the previous entry eventually (one omitting `Strategy` field at all and the other the `Type` inside the `Strategy` )?

Comment (zhouhao3) [reply baremetal/metal3remediation_manager_test.go:582]: This test case is added to test the handling of the case where `Strategy` is nil. If there is no code added in the last commit, there will be a panic here.

Comment (furkatgofurov7) [reply baremetal/metal3remediation_manager_test.go:582]: That is true, sorry I was wrong, this is a different test case actually

**Review by furkatgofurov7 — APPROVED** [review]
/approve

**Review by kashifest — COMMENTED** [review]
/lgtm

**Pull request merged**